feat: implemented dynamic import optimization for app store services

Vansh5632 opened the pull request:
## What does this PR do?

- Add dynamic import service maps for calendars, videos, and payments
- Generate CalendarServiceMap, VideoAdapterMap, and PaymentServiceMap
- Update core loading functions to use lazy loading
- Optimize bundle size and improve page load performance
- Add constants file for service configuration

- Fixes #23104  (GitHub issue number)
- Fixes CAL-6255 (Linear issue number - should be visible at the bottom of the GitHub issue description)

#### Video Demo (if applicable):

https://github.com/user-attachments/assets/8fe0e3f6-eb0c-417e-a772-5fbb75dcce4f

#### Image Demo (if applicable):
before:
<img width="279" height="309" alt="Screenshot from 2025-08-23 14-59-42" src="https://github.com/user-attachments/assets/21ab0ce6-8484-42a3-be88-c24bab546d9c" />
after:
<img width="305" height="329" alt="Screenshot from 2025-08-26 20-03-39" src="https://github.com/user-attachments/assets/b679967f-eb12-47f5-96b8-937fd45cfe4f" />

/claim #23104

keithwillcode commented:
@Vansh5632 This is looking quite good because we know that the video adapters and payment apps are going to help a lot when separated. My PR #22450 was just merged and caused a lot of conflicts here since it build the calendars map. Can you take a look and fix those and then I'll review here? Thanks!

Vansh5632 commented:
okay i will do this and make the necessary changes

Vansh5632 commented:
@keithwillcode now i will focus on the working of the video adapters

Vansh5632 committed:
feat: implement dynamic imports for video adapters

- Optimize video adapter loading with dynamic imports for better performance
- Generate VideoAdapterMap for conditional loading of video adapters
- Update videoClient.ts to use dynamic import pattern with proper error handling
- Add comprehensive mock system for video adapters in test environment
- Support for 11 video adapters: dailyvideo, huddle01video, jelly, jitsivideo, nextcloudtalk, office365video, shimmervideo, sylapsvideo, tandemvideo, webex, zoomvideo
- Maintain backward compatibility while improving application performance
- Use NEXT_PUBLIC_IS_E2E for E2E test compatibility instead of PLAYWRIGHT env var

Vansh5632 committed:
Merge branch 'main' into feat/dynamic-imports-optimization

keithwillcode commented:
/tip 100 @Vansh5632
